aiokafka/conn.py

```diff
@@ -17,21 +24,36 @@ def create_conn(host, port, *, loop=None):
 class AIOKafkaConnection:
     HEADER = struct.Struct('>i')
 
-    def __init__(self, host, port, *, loop):
+    DEFAULT_CONFIG = {
+        'client_id': 'kafka-python-' + __version__,
+        'request_timeout_ms': 40000,
+        'api_version': (0, 8, 2),  # default to most restrictive
+    }
+
+    log = logging.getLogger(__name__)
+
+    def __init__(self, host, port, *, loop, **config):
         self._host = host
         self._port = port
         self._reader = self._writer = None
         self._loop = loop
         self._requests = []
         self._read_task = None
+        self._correlation_id = 0
+        self._config = copy.copy(self.DEFAULT_CONFIG)
+        for key in self._config:
+            if key in config:
+                self._config[key] = config[key]
 
+    @asyncio.coroutine
     def _connect(self):
-        self._reader, self._writer = yield from asyncio.open_connection(
-            self.host, self.port, loop=self._loop)
-        self._read_task = asyncio.async(self._read(), loop=self._loop)
+        future = asyncio.open_connection(self.host, self.port, loop=self._loop)
+        self._reader, self._writer = yield from asyncio.wait_for(
+            future, self._config['request_timeout_ms']/1000)
+        self._read_task = asyncio.ensure_future(self._read(), loop=self._loop)
```

Comment:
We can add something like https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/helpers.py#L17-L20